### PR TITLE
#163388100 Get Articles Feature

### PR DESCRIPTION
**Pull request by asirvex**
#### What does this PR do?
It enables a user to create an article
#### Description of Task to be completed?

 - Create the list article component
 - Add the functionality for list articles
 - Create the get an article component
 - Add the functionality for getting an article

#### How should this be manually tested?
1. Clone the repo
```
git clone https://github.com/andela/ah-the-phoenix-frontend.git
```
2. Change it into the current directory
```
cd ah-the-phoenix-frontend
```
3. Checkout to the feature branch
```    
git checkout ft-get-article-163388100
```
4. Install the required dependencies using 
```
npm install
```
5. Run the server
```
npm start
```
6. Login to the website then go to the following urls
```
/articles
/articles/<article-slug>
```

Any background context you want to provide?
N/A

What are the relevant pivotal tracker stories?
[#163388100](https://www.pivotaltracker.com/story/show/163388100)

### CODE REVIEW

**Comment by ezrqnkemboi** [comment]
Squash one commit @asirvex.

File: package.json [hunk 1]
```diff
@@ -7,13 +7,14 @@
...
   "dependencies": {
+    "axios": "^0.18.0",
     "eslint": "5.12.0",
     "eslint-plugin-flowtype": "^3.4.2",
     "eslint-plugin-import": "^2.16.0",
     "eslint-plugin-jsdoc": "^4.1.0",
     "eslint-plugin-jsx-a11y": "^6.2.1",
-    "axios": "^0.18.0",
     "firebase": "^5.8.6",
+    "lorem-ipsum": "^2.0.0",
```
Comment (ezrqnkemboi) [inline package.json]: Why are we using this?

File: src/components/Articles/Article.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React, {Component} from 'react'
+import {getArticle} from "../../redux/actioncreators/getArticleActions"
+import {connect} from 'react-redux'
+import {Segment, Loader, Container} from 'semantic-ui-react';
+
+export class Article extends Component {
+    componentDidMount() {
+        this
+            .props
+            .getArticle()
```
Comment (ezrqnkemboi) [inline src/components/Articles/Article.js]: This is not good practice. 
It should be `this.props.getArticle()` in one line, for good readability.

File: src/components/Articles/ListArticles.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    Image
+} from "semantic-ui-react";
+import {axiosWithToken, axiosDefault} from '../../utils/axios_config'
+import {connect} from 'react-redux'
+import {getArticles} from "../../redux/actioncreators/listArticlesActions";
+
+export class ListArticles extends Component {
+
+    componentDidMount = () => {
+        this
```
Comment (ezrqnkemboi) [inline src/components/Articles/ListArticles.js]: Make this readable.

File: src/components/Articles/ListArticles.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            .props
+            .getArticles()
+    }
+
+    getWords = str => {
+        return str
+            .split(/\s+/)
```
Comment (ezrqnkemboi) [inline src/components/Articles/ListArticles.js]: We can make this better too.

File: src/components/Articles/Article.js (hunk 2 above)
Comment (asirvex) [reply src/components/Articles/Article.js]: Okay, let me correct that. The beautify plugin changed it to that on save

**Comment by Arusey** [comment]
@asirvex consider getting articles on the profiles route. It allows users to see what articles have been created by the current user

**Comment by ezrqnkemboi** [comment]
@asirvex, have you worked on @Arusey suggestion or have you had sync with him to get more clarification?

File: src/redux/actiontypes.js [hunk 5]
```diff
@@ -30,3 +30,11 @@ export const PROFILE_FAILURE = 'PROFILE_FAILURE';
 export const EDIT_PROFILE_REQUEST = 'EDIT_PROFILE_REQUEST';
 export const EDIT_PROFILE_SUCCESS = 'EDIT_PROFILE_SUCCESS';
 export const EDIT_PROFILE_FAILURE = 'EDIT_PROFILE_FAILURE';
+
+export const GET_ARTICLES_REQUEST = 'GET_ARTICLES_REQUEST';
+export const GET_ARTICLES_SUCCESS = 'GET_ARTICLES_SUCCESS';
+export const GET_ARTICLES_FAILURE = 'GET_ARTICLES_FAILURE';
+
+export const GET_ARTICLE_REQUEST = 'GET_ARTICLE_REQUEST';
```
Comment (ezrqnkemboi) [inline src/redux/actiontypes.js]: Rename ```GET_SINGLE_ARTICLE_REQUEST```

**Pull request merged**